### PR TITLE
1061: generate multi-architecture (amd64, arm64) docker image

### PR DESCRIPTION
**Pull request by denisa**
see https://github.com/denisa/terrascan/pkgs/container/terrascan

Note that the `docker buildx build` must build & push the image in a single operation, so I had to add some logic to the docker-build.sh script to support:
- building locally an image for the local computer architecture only
- building a multi-architecture image on CI

This PR also partially resolves warnings related to the deprecation of node.js 12; I did not attempt to update `codecov/codecov-action@v1` as the most recent (v3) version is not compatible and I would not be able to test it.

### CODE REVIEW

**Comment by denisa** [comment]
@tenable/terrascan-maintainers … any comments?

**Comment by denisa** [comment]
@bkizer-tenable: any comment? Can this be merged?

**Comment by denisa** [comment]
@Rchanger , @srokade-tenable ?

**Comment by Rchanger** [comment]
> @Rchanger , @srokade-tenable ?

sorry for late reply let me review it once and will get back,thank you for your efforts.

File: .github/workflows/gobuild.yml [hunk 1]
```diff
@@ -58,24 +58,26 @@ jobs:
   push:
     # Ensure "validate" job passes before pushing image.
     needs: validate
+    permissions:
```
Comment (Rchanger) [inline .github/workflows/gobuild.yml:61]: why do we need this permission I dont see we are writing anything

**Review by Rchanger — APPROVED** [review]

**Comment by denisa** [comment]
@Rchanger thank you for the approval.
How will this be merged? I can’t do it myself.

**Comment by Rchanger** [comment]
> @Rchanger thank you for the approval. How will this be merged? I can’t do it myself.
I'll merge.

**Pull request merged**